### PR TITLE
chore(package): update webpack to version 4.1.0

### PR DESCRIPTION
**Pull request by jaebradley**
Closes #154 and #158 

@thallada I tried to follow as much of [edx/front-end-cookie-cutter-application#11](https://github.com/edx/front-end-cookie-cutter-application/pull/11/) as possible.

Let's work off of this PR since it's the latest webpack 4 version (~~I take that back, `4.1.1` got published yesterday `#lesigh`~~ should now be updated to `4.1.1`).

What's the right way to make sure our builds haven't regressed in some way? Install Paragon from this commit hash in SFE & edX Platform?

[Here's an article about webpack 4 that was published by the webpack core team.](https://medium.com/webpack/webpack-4-released-today-6cdb994702d4)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -72,7 +72,7 @@
     "eslint-plugin-compat": "^2.1.0",
     "eslint-plugin-jsx-a11y": "^5.1.0",
     "eslint-plugin-react": "^7.1.0",
-    "extract-text-webpack-plugin": "^3.0.1",
+    "extract-text-webpack-plugin": "^4.0.0-beta.0",
```
Comment (jaebradley) [inline package.json]: I prefer specifying a semver value instead of `next` if we can help it.

File: package.json [hunk 2]
```diff
@@ -90,8 +90,9 @@
     "style-loader": "^0.20.1",
     "travis-deploy-once": "^4.3.2",
     "uglifyjs-webpack-plugin": "^1.1.8",
-    "webpack": "^3.0.0",
+    "webpack": "^4.1.1",
     "webpack-build-notifier": "^0.1.22",
+    "webpack-cli": "^2.0.10",
```
Comment (jaebradley) [inline package.json]: `webpack-cli` is necessary for webpack 4

File: webpack.config.js [hunk 3]
```diff
@@ -1,5 +1,4 @@
 const path = require('path');
-const UglifyJsPlugin = require('uglifyjs-webpack-plugin');
```
Comment (arizzitano) [inline webpack.config.js]: Does Webpack 4 handle minification/uglification out of the box?

**Comment by arizzitano** [comment]
Just to sanity check that this works, I'd recommend installing a copy of Paragon built from this branch into edx-platform and/or SFE. @thallada suggested adding a `prepare` script to run `npm run build` -- if we do that, we can just install directly from this branch without having to check in any dist files (i.e. `"paragon": "edx/paragon/greenkeeper/webpack-4.1.0"`)

**Comment by thallada** [comment]
I'm having some trouble running `npm start` on this branch. I think we should hold off on this change until [storybook supports Webpack 4](https://github.com/storybooks/storybook/pull/3148).

Once that happens. How do you all feel about using webpack-merge like we do in other projects so that we could have a separate `webpack.dev.config.js`, `webpack.dev.config.js`, and `webpack.prod.config.js`? I don't think we need a separate themeable and static build for development so it might speed things up a little.

**Comment by thallada** [comment]
Re: https://github.com/edx/paragon/pull/187, we should go straight to webpack 4.2.0.

It looks like storybook's webpack 4 support might be released as an alpha soon.

**Comment by thallada** [comment]
Storybook has released an alpha with Webpack 4 support to NPM https://www.npmjs.com/package/@storybook/react We can now install it by specifying `"alpha"` in the `package.json`.

**Pull request closed without merging**